bitcoin/src/consensus/params.rs

```diff
@@ -49,7 +49,80 @@ pub struct Params {
+            rule_change_activation_threshold: 1916, // 95%
+            miner_confirmation_window: 2016,
+            pow_limit: Target::MAX_ATTAINABLE_MAINNET,
+            pow_target_spacing: 10 * 60,            // 10 minutes.
+            pow_target_timespan: 14 * 24 * 60 * 60, // 2 weeks.
+            allow_min_difficulty_blocks: false,
+            no_pow_retargeting: false,
+        }
+    }
+
+    /// Creates testnet parameters.
+    pub const fn testnet() -> Self {
+        Params {
+            network: Network::Testnet,
+            bip16_time: 1333238400,                 // Apr 1 2012
+            bip34_height: 21111, // 0000000023b3a96d3484e5abb3755c413e7d41500f8e2a5c3f0dd01299cd8ef8
+            bip65_height: 581885, // 00000000007f6655f22f98e72ed80d8b06dc761d5da09df0fa1dc4be4f861eb6
+            bip66_height: 330776, // 000000002104c8c45e99a8853285a3b592602a3ccde2b832481da85e9e4ba182
+            rule_change_activation_threshold: 1512, // 75%
+            miner_confirmation_window: 2016,
+            pow_limit: Target::MAX_ATTAINABLE_TESTNET,
+            pow_target_spacing: 10 * 60,            // 10 minutes.
+            pow_target_timespan: 14 * 24 * 60 * 60, // 2 weeks.
+            allow_min_difficulty_blocks: true,
+            no_pow_retargeting: false,
+        }
+    }
+
+    /// Creates signet parameters.
+    pub const fn signet() -> Self {
+        Params {
+            network: Network::Signet,
+            bip16_time: 1333238400, // Apr 1 2012
+            bip34_height: 1,
+            bip65_height: 1,
+            bip66_height: 1,
+            rule_change_activation_threshold: 1916, // 95%
+            miner_confirmation_window: 2016,
+            pow_limit: Target::MAX_ATTAINABLE_SIGNET,
+            pow_target_spacing: 10 * 60,            // 10 minutes.
+            pow_target_timespan: 14 * 24 * 60 * 60, // 2 weeks.
+            allow_min_difficulty_blocks: false,
+            no_pow_retargeting: false,
+        }
+    }
+
+    /// Creates regtest parameters.
+    pub const fn regtest() -> Self {
+        Params {
+            network: Network::Regtest,
+            bip16_time: 1333238400,  // Apr 1 2012
+            bip34_height: 100000000, // not activated on regtest
+            bip65_height: 1351,
+            bip66_height: 1251,                    // used only in rpc tests
+            rule_change_activation_threshold: 108, // 75%
+            miner_confirmation_window: 144,
+            pow_limit: Target::MAX_ATTAINABLE_REGTEST,
+            pow_target_spacing: 10 * 60,            // 10 minutes.
+            pow_target_timespan: 14 * 24 * 60 * 60, // 2 weeks.
+            allow_min_difficulty_blocks: true,
+            no_pow_retargeting: true,
+        }
+    }
+
+    /// Creates parameters set for the given network.    /// Creates parameters set for the given network.
+    #[deprecated(since = "TBD", note = "use mainnet, testnet etc. instead")]
```

Comment:
I don't think we should jump on this just yet.

Reply:
Deprecation has been removed.